api/src/main/java/org/openmrs/api/db/hibernate/HibernateContextDAO.java

```diff
@@ -226,6 +235,25 @@ public User getUserByUuid(String uuid) {
 		return u;
 	}
 	
+	/**
+	 * @see org.openmrs.api.db.ContextDAO#getUserByUsername(String)
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public User getUserByUsername(String username) {
+		return userDao.getUserByUsername(username);
+	}
+	
+	/**
+	 * @throws Exception 
+	 * @see org.openmrs.api.db.ContextDAO#createUser(User, String)
+	 */
+	@Override
+	@Transactional
+	public User createUser(User user, String password, List<String> roleNames) throws Exception {
+		return Daemon.createUser(user, password, roleNames);
```

Comment:
Do we have a test for this?

Reply:
Well it is in fact already tested in `DaemonTest`, but also `Context.getContextDAO()` is not accessible from `ContextDAOTest` as this method has the package-only visibility and
- `Context` is in `org.openmrs.api.context`, while
- `ContextDAO` is in `org.openmrs.api.db`.

Do we really need to test it again, and if yes, what would you suggest to workaround the package visibility issue?

Reply:
In that case there is no need to test it again. 😊